src/lib/numbers.ts

```diff
@@ -197,12 +226,13 @@ export function formatTokenAmount(
       return undefined;
     }
   }
+  const exceedingInfo = checkPriceThresholdExceeding(amount, tokenDecimals);
 
   const formattedAmount = showAllSignificant
-    ? formatAmountFree(amount, tokenDecimals, tokenDecimals)
-    : formatAmount(amount, tokenDecimals, displayDecimals, useCommas);
+    ? formatAmountFree(exceedingInfo?.value ?? amount, tokenDecimals, tokenDecimals)
```

Comment:
i think formatAmountFee should be unchanged, because it is usually used with the flag "showAllSignificant" to just show, for example, values like 0.00123222 in inputs

Reply:
Updated!